packages/datadog-plugin-mocha/src/index.js

```diff
@@ -139,130 +126,51 @@ function createWrapRunTests (tracer, testEnvironmentMetadata, sourceRoot) {
           return
         }
         skippedTests.add(test)
-        const { childOf, resource, ...testSpanMetadata } = getTestSpanMetadata(tracer, test, sourceRoot)
 
-        const testSpan = tracer
-          .startSpan('mocha.test', {
-            childOf,
-            tags: {
-              [SPAN_TYPE]: 'test',
-              [RESOURCE_NAME]: resource,
-              ...testSpanMetadata,
-              ...testEnvironmentMetadata,
-              [TEST_STATUS]: 'skip'
-            }
-          })
-        testSpan.context()._trace.origin = CI_APP_ORIGIN
+        const testSpan = this.startTestSpan(test)
 
+        testSpan.setTag(TEST_STATUS, 'skip')
         testSpan.finish()
       })
-    }
-  }
-}
+    })
 
-const nameToParams = {}
+    this.addSub('ci:mocha:hook-error', ({ test, error }) => {
+      const testSpan = this.startTestSpan(test)
+      testSpan.setTag(TEST_STATUS, 'fail')
+      testSpan.setTag('error', error)
+      testSpan.finish()
+    })
 
-function wrapMochaEach (mochaEach) {
-  return function mochaEachWithTrace () {
-    const [params] = arguments
-    const { it, ...rest } = mochaEach.apply(this, arguments)
-    return {
-      it: function (name) {
-        nameToParams[name] = params
-        it.apply(this, arguments)
-      },
-      ...rest
-    }
+    this.addSub('ci:mocha:mocha-each', ({ name, params }) => {
+      this._testNameToParams[name] = params
```

Comment:
Any way to store this somewhere that it can be garbage collected when the test is done?

Reply:
Or any way to not need this at all?

Reply:
not that I can think of sadly. We can't grasp the test itself in here yet

Reply:
Isn't the test available on both publisher? Maybe we could use it to store a weak map?

Reply:
it isn't available on the `mocha-each` publisher. I don't know if we can use a weak map here, the keys are strings (test names)